test/common/testvm.py

```diff
@@ -259,7 +273,8 @@ def _kill_ssh_master(self):
             self.ssh_master = None
         if self.ssh_process:
             self.ssh_process.stdin.close()
-            self.ssh_process.wait()
+            if not self._abort:
+                self.ssh_process.wait()
```

Comment:
> That's also the case on non-timeout failures. My question is why the timeout is different from other failures with regards to downloading the logs.

This part of the timeout is specifically for aborting long-running shell commands that are executed on the test machine. If they don't terminate, the ssh master connection will remain active and the test will be left hanging until the entire test run process times out.